app/models/inat_import.rb

```diff
@@ -41,7 +54,52 @@ def add_response_error(error)
     save
   end
 
+  # Users who can import others users' iNat observations
   def self.super_importers
     Project.find_by(title: "SuperImporters").user_group.users
   end
+
+  def total_expected_time
+    importables * initial_avg_import_seconds
+  end
+
+  def initial_avg_import_seconds
+    if user_import_history?
+      personal_initial_avg_import_seconds
+    elsif system_import_history?
+      system_initial_avg_import_seconds
+    else
+      BASE_AVG_IMPORT_SECONDS
+    end
+  end
+
+  def reset_last_obs_start
+    update(last_obs_start: Time.now.utc)
+  end
+
+  def last_obs_elapsed_time
+    return 0 unless last_obs_start
+
+    Time.now.utc - last_obs_start
+  end
```

Comment:
@mo-nathan: Was thinking about handling this via a small `StopWatch` class with `elapsed_time` and `reset` methods.
Reactions?

Reply:
Sounds worth trying, but not urgent.